metrics/prometheus.go

```diff
@@ -0,0 +1,172 @@
+package metrics
+
+import (
+	"github.com/prebid/prebid-cache/config"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+/**************************************************
+ *	Still missing:
+ **************************************************
+ *	> Register methods for Prometheus metrics in metrics/prometheus.go
+ *	> Modify all histograms
+ *	> Tests for config.go
+ *	> Tests for metrics/prometheus.go
+ *	> Upload so they can see progress
+ *	> Make the option betewwn Prometheus and InfluxDB configurable
+ **************************************************/
+/**************************************************
+ *	Object definition
+ **************************************************/
+
+type PrometheusMetrics struct {
+	Registry        *prometheus.Registry
+	Puts            *PrometheusMetricsEntry
+	Gets            *PrometheusMetricsEntry
+	PutsBackend     *PrometheusMetricsEntryByFormat
+	GetsBackend     *PrometheusMetricsEntry
+	Connections     *PrometheusConnectionMetrics
+	ExtraTTLSeconds *prometheus.HistogramVec
+}
+type PrometheusMetricsEntry struct {
+	Duration   *prometheus.HistogramVec
+	Errors     prometheus.Counter
+	BadRequest prometheus.Counter
+	Request    prometheus.Counter
+}
+
+type PrometheusMetricsEntryByFormat struct {
+	Duration       metrics.Histogram
+	Errors         prometheus.Counter
```

Comment:
Again any reason for not using vectors?

Reply:
Everything was refactored to make better use of the `CounterVec` objects. Please let me know what you think of this new approach. I can always roll back.